### PR TITLE
ContextUtils

### PR DESCRIPTION
**Pull request by mwear**
See #201 for discussion and rationale. I made that as a separate issue as I think there could be a larger discussion. 

This PR contains an implementation of the ContextUtils proposal to add convenience methods to operate on a context and also applies in the codebase where applicable.

### CODE REVIEW

**Comment by mwear** [comment]
I think #108 could benefit from some of the convenience methods introduced here.

File: api/lib/opentelemetry/context_utils.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  # The ContextUtils module contains convenience methods for inserting and
```
Comment (fbogsany) [inline api/lib/opentelemetry/context_utils.rb]: Missing a word in the comment here? `inserting and ...`?

**Comment by fbogsany** [comment]
So far, this doesn't feel like much of an improvement to me, probably because the methods are stuck in the `ContextUtils` class. There is probably a more convenient place for them.

Higher level concern: it feels like we've exposed a lot more of the internals of the API than I recall, especially the ability to set and get the current span and extracted span context. This is more of a "spidey sense" thing than anything concrete, but it does feel like the API is becoming a lot more complex (both in implementation and interface) due to various indirections.

**Comment by mwear** [comment]
Good points. The convenience methods for working with a span context are only useful if you're writing a propagator. I think we need some help around reading a span from a context, or returning a new context with an updated span, but those probably have a better place. We can discuss where that might be.

I'll close this and we can discuss in the related issue.

**Pull request closed without merging**